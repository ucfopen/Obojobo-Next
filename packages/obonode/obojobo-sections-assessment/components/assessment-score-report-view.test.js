import React from 'react'
import renderer from 'react-test-renderer'

import ReportView from './assessment-score-report-view'

const TYPE_TEXT = 'text'
const TYPE_DIV = 'divider'
const TYPE_EXTRA_CRED = 'extra-credit'
const TYPE_PENALTY = 'penalty'
const TYPE_VAL = 'value'
const TYPE_TOTAL = 'total'
const VAL_DID_NOT_PASS = 'Did Not Pass'

describe('AssessmentScoreReportView', () => {
	beforeEach(() => {
		jest.spyOn(Math, 'random').mockReturnValue(0)
	})

	test('AssessmentScoreReportView component', () => {
		const report = {
			textItems: [],
			scoreChangeDescription: null
		}

		const component = renderer.create(<ReportView report={report} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('AssessmentScoreReportView component with scoreChangeDescription', () => {
		const report = {
			textItems: [],
			scoreChangeDescription: 'mockDescription'
		}

		const component = renderer.create(<ReportView report={report} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('AssessmentScoreReportView component with textItems', () => {
		const report = {
			textItems: [
				{
					type: TYPE_TEXT,
					text: 'mockText'
				},
				{
					type: TYPE_DIV
				},
				{
					type: TYPE_EXTRA_CRED,
					value: '5'
				},
				{
					type: TYPE_PENALTY,
					value: '5'
				},
				{
					type: TYPE_VAL,
					value: '100'
				},
				{
					type: TYPE_TOTAL,
					value: '100'
				}
			],
			scoreChangeDescription: 'mockDescription'
		}

		const component = renderer.create(<ReportView report={report} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('AssessmentScoreReportView component did not pass', () => {
		const report = {
			textItems: [
				{
					type: TYPE_VAL,
					value: VAL_DID_NOT_PASS
				}
			],
			scoreChangeDescription: 'mockDescription'
		}

		const component = renderer.create(<ReportView report={report} />)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})
})
