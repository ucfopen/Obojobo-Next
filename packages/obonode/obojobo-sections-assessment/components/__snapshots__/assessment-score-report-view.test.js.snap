// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentScoreReportView AssessmentScoreReportView component 1`] = `
<div
  className="obojobo-draft--sections--assessment--components--score-report"
>
  <div
    className="text-items"
  />
</div>
`;

exports[`AssessmentScoreReportView AssessmentScoreReportView component did not pass 1`] = `
<div
  className="obojobo-draft--sections--assessment--components--score-report"
>
  <div
    className="text-items"
  >
    <div
      className="value"
    >
      <div
        className="label"
      />
      <span
        className="amount is-null"
      >
        Did Not Pass
      </span>
    </div>
  </div>
  <span
    className="score-change-description"
  >
    mockDescription
  </span>
</div>
`;

exports[`AssessmentScoreReportView AssessmentScoreReportView component with scoreChangeDescription 1`] = `
<div
  className="obojobo-draft--sections--assessment--components--score-report"
>
  <div
    className="text-items"
  />
  <span
    className="score-change-description"
  >
    mockDescription
  </span>
</div>
`;

exports[`AssessmentScoreReportView AssessmentScoreReportView component with textItems 1`] = `
<div
  className="obojobo-draft--sections--assessment--components--score-report"
>
  <div
    className="text-items"
  >
    <div
      className="text"
    >
      mockText
    </div>
    <hr
      className="divider"
    />
    <div
      className="extra-credit"
    >
      <span
        className="label"
      >
        <span>
          Extra-credit
        </span>
         - 
      </span>
      <span
        className="amount is-number"
      >
        +5
        %
      </span>
    </div>
    <div
      className="penalty"
    >
      <span
        className="label"
      >
        <span>
          Penalty
        </span>
         - 
      </span>
      <span
        className="amount is-number"
      >
        -5
        %
      </span>
    </div>
    <div
      className="value"
    >
      <div
        className="label"
      />
      <span
        className="amount is-number"
      >
        100
        %
      </span>
    </div>
    <div
      className="total"
    >
      <div
        className="label"
      />
      <div
        className="amount is-number"
      >
        100
        %
        <span
          className="great-job-you-rock"
        >
          ðŸ˜Ž
        </span>
      </div>
    </div>
  </div>
  <span
    className="score-change-description"
  >
    mockDescription
  </span>
</div>
`;
